go/vt/vtgate/vtgate.go

```diff
@@ -194,14 +199,32 @@ func Init(ctx context.Context, serv srvtopo.Server, cell string, tabletTypesToWa
 	srvResolver := srvtopo.NewResolver(serv, gw, cell)
 	resolver := NewResolver(srvResolver, serv, cell, sc)
 	vsm := newVStreamManager(srvResolver, serv, cell)
+
+	var si SchemaInfo = nil
+	var st *vtschema.Tracker
+	if *enableSchemaChangeSignal {
+		st = vtschema.NewTracker(gw.hc.Subscribe())
+		addKeyspaceToTracker(ctx, srvResolver, st, gw)
+		si = st
+	}
+
 	cacheCfg := &cache.Config{
 		MaxEntries:     *queryPlanCacheSize,
 		MaxMemoryUsage: *queryPlanCacheMemory,
 		LFU:            *queryPlanCacheLFU,
 	}
 
+	executor := NewExecutor(ctx, serv, cell, resolver, *normalizeQueries, *warnShardedOnly, *streamBufferSize, cacheCfg, si)
```

Comment:
nit: we do not explicitly convert to SchemaInfo before calling `NewExecutor` method. `st` can be sent as well.

Reply:
you would think so, right?
you'd be wrong. :(

https://glucn.medium.com/golang-an-interface-holding-a-nil-value-is-not-nil-bb151f472cc7